lib/site_setting_extension.rb

```diff
@@ -482,7 +485,21 @@ def setup_shadowed_methods(name, value)
   def setup_methods(name)
     clean_name = name.to_s.sub("?", "").to_sym
 
-    if type_supervisor.get_type(name) == :upload
+    if type_supervisor.get_type(name) == :uploaded_image_list
+      define_singleton_method clean_name do
```

Comment:
I'm wondering why in the setting case this is happening in the controller instead of here. I understand that you followed upload, but it's not clear to me why uploads are that way either.

Reply:
What I understand that is different about `:upload` and `:uploaded_image_list` is that the SiteSetting is actually an "object" and not just a string, which I believe is the reason for the translation to happen in different place.

Reply:
I guess it just feels weird to me that you can set the setting to one thing and get something completely different back when you query it.